qiita_db/support_files/patches/33.sql

```diff
@@ -449,3 +469,33 @@ DROP TABLE qiita.preprocessed_data;
 DROP TABLE qiita.raw_filepath;
 DROP TABLE qiita.raw_data;
 DROP TABLE qiita.study_pmid;
+
+-- Rename the id columns from the parameters tables
+ALTER TABLE qiita.processed_params_sortmerna RENAME COLUMN processed_params_id TO parameters_id;
+ALTER TABLE qiita.processed_params_uclust RENAME COLUMN processed_params_id TO parameters_id;
+ALTER TABLE qiita.preprocessed_sequence_454_params RENAME COLUMN preprocessed_params_id TO parameters_id;
+ALTER TABLE qiita.preprocessed_sequence_illumina_params RENAME COLUMN preprocessed_params_id TO parameters_id;
+ALTER TABLE qiita.preprocessed_spectra_params RENAME COLUMN preprocessed_params_id TO parameters_id;
+
+-- Create a function to return the roots of an artifact, i.e. the source artifacts
+CREATE FUNCTION qiita.find_artifact_roots(a_id bigint) RETURNS SETOF bigint AS $$
```

Comment:
That's a really good point - I found my-self debugging this later when I was working in the artifact object, so having specific tests would've been better!